src/general/ModLoader.cs

```diff
@@ -0,0 +1,320 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using Godot;
+using Newtonsoft.Json;
+using Directory = System.IO.Directory;
+using File = System.IO.File;
+
+public class ModLoader : Control
```

Comment:
Should be documented as well.

And it'd be nice to split the logic and UI, so that there is a ModLoader (perhaps an autoload, combined with the autoload mod loader), and a separate ModManagerUI that inherits Control.